File: src/main/scala/fortress/msfol/Names.java
package fortress.msfol;

import java.util.Set;
import java.util.HashSet;
import java.util.Arrays;

public class Names {
    private static Set<String> illegalNames = new HashSet<>(Arrays.asList(
        "and", "or", "not", "=>", "<=>", "iff",
        "forall", "exists", "=", "==",
        "true", "false", 
        "if", "else"
    ));
    
    public static boolean isIllegal(String name) {
        return illegalNames.contains(name.toLowerCase());
    }
}

File: src/main/scala/fortress/msfol/Names.scala
package fortress.msfol

object Names {
    private val illegalNames: Set[String] = Set(
        "and", "or", "not", "=>", "<=>", "iff",
        "forall", "exists", "=", "==",
        "true", "false", 
        "if", "else"
    )
    
    private val illegalPrefixes: Set[String] = Set(
        DomainElement.prefix
    )
    
    def isIllegal(name: String): Boolean =
        (illegalNames contains name.toLowerCase) || (illegalPrefixes exists (name startsWith _))
}
